### PR TITLE
[#4768] Added client API whitelist and lots of other stuff (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: lib/core/src/apiHandler.cpp [hunk 1]
```diff
@@ -109,33 +102,45 @@ namespace irods {
             return PASS( ret );
         }
 
+        namespace fs = boost::filesystem;
+
+        static std::vector<fs::path> previously_loaded_client_plugin_names;
+        static std::vector<fs::path> previously_loaded_server_plugin_names;
+        auto& loaded_plugin_names = _cli_flg ? previously_loaded_client_plugin_names
```
Comment (trel) [inline lib/core/src/apiHandler.cpp]: Codacy Issue found: [The scope of the variable 'loaded_plugin_names' can be reduced.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5204898)

Comment (trel) [inline lib/core/src/apiHandler.cpp]: Codacy Issue found: [The scope of the variable 'loaded_plugin_names' can be reduced.](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5204898)

**Comment by korydraughn** [comment]
CI tests passed on Ubuntu 16.
Did not test on Centos.

**Comment by korydraughn** [comment]
CI core tests passed on Centos 7.
Added #

**Pull request merged**